examples/cifar10/train_benchmark.py

```diff
@@ -0,0 +1,61 @@
+"""Benchmark for the CIFAR10 example."""
+import tempfile
+
+import time
+from absl import flags
+from absl.testing import absltest
+from absl.testing.flagsaver import flagsaver
+
+import jax
+import numpy as np
+from flax.testing import Benchmark
+
+import train
+
+
+# Parse absl flags test_srcdir and test_tmpdir.
+jax.config.parse_flags_with_absl()
+
+
+FLAGS = flags.FLAGS
+
+
+class CifarTenBenchmark(Benchmark):
+  """Benchmarks for the CIFAR10 Flax example."""
+
+  @flagsaver
+  def test_1x_v100(self):
+    """Run Wide ResNet CIFAR10 on 1x V100 GPUs for 2 epochs."""
+    model_dir = tempfile.mkdtemp()
+    FLAGS.num_epochs = 2
+    FLAGS.arch = 'wrn26_10'
+    FLAGS.model_dir = model_dir
+
+    start_time = time.time()
+    train.main([])
+    benchmark_time = time.time() - start_time
+    summaries = self.read_summaries(model_dir)
+
+    # Summaries contain all the information necessary for the regression
+    # metrics.
+    wall_time, _, eval_error_rate = zip(*summaries['eval_error_rate'])
+    wall_time = np.array(wall_time)
+    sec_per_epoch = np.mean(wall_time[1:] - wall_time[:-1])
```

Comment:
Is `sec_per_epoch` basically the same as `benchmark_time / len(wall_time)`?

Reply:
I think so, almost. However, the first epoch has a few one-off things happening in it, such as data caching or XLA compilation. Not taking it into account allows for a better estimate of the actual compute efficiency.

Reply:
Got it, thanks.

A thought (please don't block this work on it): If we also want to measure start-up/teardown time (which I guess is also important incase, say JIT compilation time becomes slower in JAX), we could compute:

benchmark_time - (wall_time[-1] - wall_time[0])

Which is maybe off by some constant but should be (may be?) consistent across runs.

This is probably not worth doing... Just thought I'd share.

Reply:
I think avital@ has a fair point of considering measurements of setup and teardown differently since it captures important details like JAX compilation and caching. I can actually see how that information could be very helpful in the future. 

May be this could be a feature which one of us could add in the future.

Reply:
Agreed, we should track it - it's free extra information.